src/plugins/Transloadit/index.js

```diff
@@ -173,6 +180,37 @@ module.exports = class Transloadit extends Plugin {
     return this.opts.waitForEncoding || this.opts.waitForMetadata
   }
 
+  /**
+   * Used when `importFromUploadURLs` is enabled: reserves all files in
+   * the assembly.
+   */
+  reserveFiles (fileIDs) {
+    return Promise.all(fileIDs.map((fileID) => {
+      const file = this.core.getState().files[fileID]
+      const assemblyId = file.transloadit.assembly
+      const assemblies = this.state.assemblies
+      return this.client.reserveFile(assemblies[assemblyId], file)
+    }))
+  }
+
+  /**
+   * Used when `importFromUploadURLs` is enabled: adds files to the assembly
+   * once they have been fully uploaded.
+   */
+  onFileUploadURLAvailable (fileID) {
+    const file = this.core.getState().files[fileID]
+    if (!file || !file.transloadit || !file.transloadit.assembly) {
+      return
+    }
+
+    const assembly = this.state.assemblies[file.transloadit.assembly]
+
+    this.client.addFile(assembly, file).catch((err) => {
+      this.core.log(err)
+      this.core.emit('core:upload-error', file.id, err)
```

Comment:
Hmm, this happens _after_ `core:upload-success`, it should emit this error in the postprocess step instead.